### PR TITLE
[DC-2092] Update Client instantiations with correct scopes

### PR DESCRIPTION
**Pull request by nishanthpp93**
* [DC-2092] Use function to define default creds

[DC-2092]: https://precisionmedicineinitiative.atlassian.net/browse/DC-2092?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by MichaelSchmidt82** [comment]
I looked at the PR, and I can't see any reason for issues with your changes. So I recommended some variable name changes to keep consistency. I also examined cases where the client is initiated WITHOUT setting the project_id and couldn't find any issues (it would also be a syntax error). Since the tests ALSO pass, there can't be a syntax error. Finally, all I can think to do is just run them in their respective environments for verification.

**Comment by nishanthpp93** [comment]
> I looked at the PR, and I can't see any reason for issues with your changes. So I recommended some variable name changes to keep consistency. I also examined cases where the client is initiated WITHOUT setting the project_id and couldn't find any issues (it would also be a syntax error). Since the tests ALSO pass, there can't be a syntax error. Finally, all I can think to do is just run them in their respective environments for verification.

Thanks, will do. I don't see the variable name recommendations. I'll deploy to test but we cannot test in prod so hopefully it works the same way.

**Comment by MichaelSchmidt82** [comment]
Search for: `sc = StorageClient(project)` and `self.storage_client = StorageClient(self.app_id)`

> 
https://github.com/all-of-us/curation/pull/1090/files/7619e3308c22ee7dbc7dcbbe435c72561812635d#diff-8d3cca1f1adcd26ccdab8bf7883986c79b7e8624f10315602b6050ca38f83574

https://github.com/all-of-us/curation/pull/1090/files/7619e3308c22ee7dbc7dcbbe435c72561812635d#diff-f7997bb92eb93fe8416286ed4019eb35dc08e03858fae7519440c8ebd0647fa7

**Comment by nishanthpp93** [comment]
@MichaelSchmidt82 I tested and found out that this is because `get_bucket` is not something we used before. For external projects, the Service Account only has permissions to read objects, so `get_blob` works but not `get_bucket`.
FYI @lledezma

**Review by MichaelSchmidt82 — APPROVED** [review]
We came across this situation before, and I don't think I fully understood why we needed this change.  Clearly, we do, and thanks for this patch.

File: tests/integration_tests/data_steward/tools/top_heel_errors_test.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def setUp(self):
         self.app_id = app_identity.get_application_id()
         self.dataset_id = bq_utils.get_dataset_id()
         self.bucket: str = gcs_utils.get_drc_bucket()
-        self.storage_client = StorageClient()
+        self.storage_client = StorageClient(self.app_id)
```
[the remark this answers is not in the document]
Comment (nishanthpp93) [reply tests/integration_tests/data_steward/tools/top_heel_errors_test.py:67]: Yeah I can make this change.

File: data_steward/validation/participants/writers.py [hunk 2]
```diff
@@ -69,7 +67,7 @@ def write_to_result_table(project, dataset, site, match_values):
 
     # write results
     results.seek(0)
-    sc = StorageClient()
+    sc = StorageClient(project)
```
[the remark this answers is not in the document]
Comment (nishanthpp93) [reply data_steward/validation/participants/writers.py:70]: This one will need a lot of changes so this is better in a separate ticket.

**Pull request merged**